bin/gftools-push-status.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Check push status of families on Dev, Sandbox and Production servers.
+
+This script will determine whether the families listed in
+to_production.txt and to_sandbox.txt in the google/fonts dir have been
+pushed to their respective servers within an allocated date.
+
+If the files are empty, you can always use git checkout to view past
+states.
+
+
+Usage:
+
+Check push status for the past month:
+gftools push-status path/to/google/fonts
+
+Check push status for specified date:
+gftools push-status path/to/google/fonts -pd 2020-07-01
+"""
+import argparse
+from argparse import RawDescriptionHelpFormatter
+import os
+from pathlib import Path
+import gftools.fonts_public_pb2 as fonts_pb2
+from google.protobuf import text_format
+from datetime import datetime, timedelta
+import json
+import requests
+
+
+DEV_META_URL = "http://fonts-dev.sandbox.google.com/metadata/fonts"
+SANDBOX_META_URL = "http://fonts.sandbox.google.com/metadata/fonts"
+PROD_META_URL = "http://fonts.google.com/metadata/fonts"
+
+
+def get_family_metadata(url):
+    """Get family json data from a Google Fonts metadata url"""
+    # can't do requests.get("url").json() since request text starts with ")]}'"
+    info = json.loads(requests.get(url).text[5:])
+    return {i['family']: i for i in info["familyMetadataList"]}
+
+
+DEV_META = get_family_metadata(DEV_META_URL)
+SANDBOX_META = get_family_metadata(SANDBOX_META_URL)
+PROD_META = get_family_metadata(PROD_META_URL)
+
+# If no push date is added we can assume that the push has probably
+# happened in the past month
+ONE_MONTH_AGO = datetime.now() - timedelta(days=31)
+
+
+def families_from_file(fp):
+    """Convert to_sandbox.txt and to_production.txt files to a list of
+    family names."""
+    results = set()
+    with open(fp, "r") as doc:
+        family_dirs = doc.read().split()
+
+    gf_path = Path(fp).parent
+    for dir_ in family_dirs:
+        metadata_file = gf_path.joinpath(f"{dir_}/METADATA.pb")
+        if not metadata_file.is_file():
+            raise FileNotFoundError(f"No METADATA.pb file found in {dir_}")
```

Comment:
It's nice to lean on comprehensions.

Consider indicating all the bad paths, e.g.

```python
metadata_files = [Path(fp).parent / d / 'METADATA.pb' for d in family_dirs]
missing_files = [f for f in metadata_files if not f.is_file()]
...raise an error reporting all the missing ones, or the first N missing ones...

return [read_proto(f, fonts_pb2.FamilyProto()).name for f in metadata_files]
```